kapitan/refs/secrets/azkms.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2019 The Kapitan Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"gkms secrets module"
+
+import os
+import base64
+import logging
+import warnings
+
+from kapitan.refs.base64 import Base64Ref, Base64RefBackend
+from kapitan.refs.base import RefError
+from kapitan import cached
+from kapitan.errors import KapitanError
+
+from azure.keyvault.keys import KeyClient
+from azure.identity import DefaultAzureCredential
+from azure.core.exceptions import HttpResponseError
+from azure.keyvault.keys.crypto import EncryptionAlgorithm
+
+
+logger = logging.getLogger(__name__)
+
+# We are using default credentials as defined in the link:
```

Comment:
This is a good candidate have in documentation instead of comments. I'd suggest adding this and any other needed docs for azkms to `docs/secrets.md` :) That way they are also searchable/indexed in the website https://kapitan.dev

Reply:
ok